### ISSUE TITLE
Question about Accuracy for Partitioned Model

### ISSUE THREAD

**Issue by benwestermeyer** [ulfaslak/computational_analysis_of_big_data_2018_spring#6]
I tried training my model on one half of the data and then testing it on the other half, as Ulf did on the Week 4 slides. The accuracy, precision, and recall scores are all greater than .99. I'm wondering if I did something wrong or if the model is actually this accurate. Thanks!

```
import pandas as pd
from sklearn.ensemble import RandomForestClassifier
from sklearn.naive_bayes import BernoulliNB
from sklearn.metrics import accuracy_score, precision_score, f1_score, recall_score
import numpy as np

x = pd.read_csv('alliances.csv', index_col=0)
target = pd.DataFrame(x.iloc[:,-1])

x = np.array(x)
target = np.array(target)

model = RandomForestClassifier()
N = np.ravel(x.shape)[0]

trainindices = range(0, N, 2)
testindices = range(1, N, 2)
    
model.fit(x[trainindices, :], target[trainindices].ravel())

ytrue = target[testindices]
ypred = model.predict(x[testindices, :])


print accuracy_score(ytrue, ypred)
print precision_score(ytrue, ypred)
print recall_score(ytrue, ypred)
```

**Comment by ulfaslak** [comment]
I spot the issue. Your feature matrix `x` includes the target. So when you train your model, one of the features you train on IS the outcome. You basically *give away* the answer and then ask for the answer: of course you're gonna get a perfect score.

Change `x = np.array(x)` to `x = np.array(x)[:, :-1]`.
